proposals/route-blinding.md

```diff
@@ -0,0 +1,308 @@
+  E(0) = e(0) * G
+
+Blinding:
+
+  For i = 0 to r-1:
+    ss(i) = H(e(i) * N(i)) = H(k(i) * E(i))         // shared secret known only by N(r) and N(i)
+    B(i) = HMAC256("blinded_node_id", ss(i)) * N(i) // Blinded node_id for N(i), private key known only by N(i)
+    rho(i) = HMAC256("rho", ss(i))                  // Key used to encrypt payload for N(i) by N(r)
+    e(i+1) = H(E(i) || ss(i)) * e(i)                // Ephemeral private key, only known by N(r)
+    E(i+1) = H(E(i) || ss(i)) * E(i)                // NB: N(i) must not learn e(i)
+
+Blinded route:
+  
+  (P(0),fees(0),cltv(0),encrypted_data(0))
+  (B(1),fees(1),cltv(1),encrypted_data(1))
+  ...
+  (B(r-1),fees(r-1),cltv(r-1),encrypted_data(r-1))
+```
+
+Note that this is exactly the same construction as Sphinx, but at each hop we use the shared secret
+to derive a blinded `node_id` for `N(i)` for which the private key will only be known by `N(i)`.
+
+The recipient needs to provide `E(0)` and the blinded route to potential senders.
+
+The `encrypted_data(i)` is encrypted with ChaCha20-Poly1305 using the `rho(i)` key, and
+contains the real `short_channel_id` to forward to (and potentially other fields).
+
+Note that the introduction point uses the real `node_id`, not the blinded one, because the sender
+needs to be able to locate this introduction point and find a route to it. But the sender will send
+`E(0)` in the onion `hop_payload` for `N(0)`, which will allow the introduction point to compute
+the shared secret and correctly forward.
+
+Note that in the specific case of payments, the recipient can sign the invoice with `e(0)`.
+The sender will recover `E(0)` from the signature so no extra field needs to be added to Bolt 11.
+And this ensures the recipient doesn't reveal his real `node_id` through the invoice signature.
+
+However, if the recipient wants to be able to prove invoice ownership in the future, she should
+sign the invoice with a different key and provide `E(0)` via a dedicated Bolt 11 field.
+
+### Sending to a blinded route
+
+The sender finds a route to the introduction point `N(0)`, and extends it with the blinded route.
+It then creates an onion for that route, and includes `E(0)` and `encrypted_data(0)` in
+the onion payload for `N(0)`.
+
+When `N(0)` receives the onion and decrypts it, it finds `E(0)` in the payload and is able to
+compute the following:
+
+```text
+  ss(0) = H(k(0) * E(0))
+  rho(0) = HMAC256("rho", ss(0))
+  E(1) = H(E(0) || ss(0)) * E(0)
+```
+
+It uses `rho(0)` to decrypt the `encrypted_data(0)` and discover the `scid` to forward to.
+It forwards the onion to the next node and includes `E(1)` in a TLV field in the message extension
+(at the end of the `update_add_htlc` message).
+
+All the following intermediate nodes `N(i)` do the following steps:
+
+```text
+  E(i) <- extracted from update_add_htlc's TLV extension
+  ss(i) = H(k(i) * E(i))
+  b(i) = HMAC256("blinded_node_id", ss(i)) * k(i)
+  Use b(i) to decrypt the incoming onion
```

Comment:
You're right, I should make this clearer, it means you should use `b(i)` (your blinded private key) instead of your node's private key when doing the ECDH required by Sphinx.

Reply:
Done in https://github.com/lightning/bolts/pull/765/commits/5cfbfeca6e4b752d411d672008d8119c77957f1b